views/dashboard_barista.ejs

```diff
@@ -74,12 +74,89 @@
           </tr>
         <% } %>
 
-      <% if (shift.length == 0 ) { %>
+      <% if (activeShift.length == 0 ) { %>
           <tr>
             <th colspan="5" class="text-center">There is no active shift</th>
           </tr>        
       <% } %>           
     </table>
   </section>
+
+  <!-- Past shifts display section -->
+  <h2 class="text-center text-secondary text-2xl font-bold pt-8 pb-3">Fulfilled or Past shifts</h2>

```

Comment:
This is a lot of code to copy paste. I wonder if you can make a reusable component that accepts the data as an attribute https://www.digitalocean.com/community/tutorials/how-to-use-ejs-to-template-your-node-application#step-4-passing-data-to-views-and-partials

I think you could use the pattern described in the "Passing Data to a Partial in EJS" section of that link

Reply:
Add `barista_shifts.ejs` in `partials` folder as a reusable component.